### PR TITLE
Fixed bug searching by tags and deviceId at the same time.   

### PR DESCRIPTION
**Pull request by clare**

### CODE REVIEW

File: models/Recording.js [hunk 1]
```diff
@@ -193,10 +193,10 @@ module.exports = function(sequelize, DataTypes) {
 
 
   // local
-  var handleTagMode = (tagMode, tagWhats) => {
+  var handleTagMode = (tagMode, tagWhatsIn) => {
+    const tagWhats = (tagWhatsIn && (tagWhatsIn.length > 0)) ? tagWhatsIn : null;
     if (!tagMode) {
-      const hasWhatTags = (tagWhats && (tagWhats.length > 0));
-      tagMode = (hasWhatTags) ? 'tagged' : 'any';
+      tagMode = (tagWhats) ? 'tagged' : 'any';
```
Comment (gferraro) [inline models/Recording.js]: could remove brackets

File: models/Recording.js [hunk 2]
```diff
@@ -228,7 +228,7 @@ module.exports = function(sequelize, DataTypes) {
   };
 
   var tagOfType = (tagWhats, tagTypeSql) => {
-    return 'EXISTS (' + recordingTaggedWith(tagWhats, tagTypeSql) + ') OR EXISTS(' + trackTaggedWith(tagWhats, tagTypeSql) + ')';
+    return '(EXISTS (' + recordingTaggedWith(tagWhats, tagTypeSql) + ') OR EXISTS(' + trackTaggedWith(tagWhats, tagTypeSql) + '))';
   };
 
   var notTagOfType = (tagWhats, tagTypeSql) => {
```
Comment (gferraro) [inline models/Recording.js]: Should this be enclodes in brakcs as well?
```    return 'NOT EXISTS (' + recordingTaggedWith(tagWhats, tagTypeSql) + ') AND NOT EXISTS(' + trackTaggedWith(tagWhats, tagTypeSql) + ')';```

File: models/Recording.js [hunk 3]
```diff
@@ -193,10 +193,10 @@ module.exports = function(sequelize, DataTypes) {
...
-  var handleTagMode = (tagMode, tagWhats) => {
+  var handleTagMode = (tagMode, tagWhatsIn) => {
+    const tagWhats = (tagWhatsIn && (tagWhatsIn.length > 0)) ? tagWhatsIn : null;
     if (!tagMode) {
-      const hasWhatTags = (tagWhats && (tagWhats.length > 0));
-      tagMode = (hasWhatTags) ? 'tagged' : 'any';
+      tagMode = (tagWhats) ? 'tagged' : 'any';
     }
 
     const humanSQL = 'NOT "Tags".automatic';
```
Comment (gferraro) [inline models/Recording.js]: Could remove the extra brackets in lines 204 - 226
 ```return '(' + tagOfType(tagWhats, humanSQL) + ') AND (' + tagOfType(tagWhats, AISQL) + ')';```
could be
```return  tagOfType(tagWhats, humanSQL) + ' AND ' + tagOfType(tagWhats, AISQL);```

Comment (clare) [reply models/Recording.js]: Done

File: models/Recording.js (hunk 2 above)
Comment (clare) [reply models/Recording.js]: This one doesn't 'need brackets' its the 'or' clause that causes the problem.

File: models/Recording.js (hunk 1 above)
Comment (gferraro) [reply models/Recording.js]: oh sorry i meant in line 199
```tagMode = (tagWhats) ? 'tagged' : 'any';```
but thats ok

**Pull request merged**